README.md
# MailHog-debian-package

I try to package https://github.com/mailhog/MailHog to Debian/Ubuntu OS. But my first needis that it has to work on ubuntu 12.04 (aka ubuntu precise)

## Easy setup ...
Use the bintray.com debian repository. I sign my package with the default [bintray gpg key](https://bintray.com/user/downloadSubjectPublicKey?username=bintray)

```
echo "deb https://dl.bintray.com/deogracia/mailhog precise contrib" | sudo tee -a /etc/apt/sources.list
wget https://bintray.com/user/downloadSubjectPublicKey?username=bintray -O bintray.asc
sudo apt-key add bintray.asc
sudo apt-get clean
sudo apt-get update
sudo apt-get install mailhog
```
I use the mailhog defaults

## ... Or use the build instructions
You need 
 * build-install
 * fakeroot
 
```
git clone https://github.com/deogracia/MailHog-debian-package
cd MailHog-debian-package
mkdir -p package/usr/sbin
chmod 755 package/usr/sbin
cd package/usr/sbin
wget https://github.com/mailhog/MailHog/releases/download/v0.1.6/MailHog_linux_amd64
mv MailHog_linux_amd64 mailhog
chmod 755 mailhog
cd ../../..
bash restore-permission.bash

fakeroot dpkg-deb --build package

mv package.deb mailhog-VERSION-amd64.deb
```

## Known errors

### Lintian
* `new-package-should-close-itp-bug` : ignored


debian/changelog
mailhog (0.1.6-1) unstable; urgency=low

  * Initial release
  * First package

 -- Lionel FéLcité <deogracia@free.fr>  Tue, 31 Mar 2015 21:56:52 +0000


debian/prerm.ex
#!/bin/sh
# prerm script for mailhog
#
# see: dh_installdeb(1)

set -e

# summary of how this script can be called:
#        * <prerm> `remove'
#        * <old-prerm> `upgrade' <new-version>
#        * <new-prerm> `failed-upgrade' <old-version>
#        * <conflictor's-prerm> `remove' `in-favour' <package> <new-version>
#        * <deconfigured's-prerm> `deconfigure' `in-favour'
#          <package-being-installed> <version> `removing'
#          <conflicting-package> <version>
# for details, see http://www.debian.org/doc/debian-policy/ or
# the debian-policy package


case "$1" in
    remove|upgrade|deconfigure)
    ;;

    failed-upgrade)
    ;;

    *)
        echo "prerm called with unknown argument \`$1'" >&2
        exit 1
    ;;
esac

# dh_installdeb will replace this with shell code automatically
# generated by other debhelper scripts.

#DEBHELPER#

exit 0


debian/init.d.ex
#!/bin/sh
### BEGIN INIT INFO
# Provides:          mailhog
# Required-Start:    $network $local_fs
# Required-Stop:
# Default-Start:     2 3 4 5
# Default-Stop:      0 1 6
# Short-Description: Web and API based SMTP testing 
# Description:       Inspired by MailCatcher, easier to install.
#                    Built with Go - MailHog runs without installation on multiple platforms.
#                    MailHog is an email testing tool for developers
### END INIT INFO

# Author: Lionel Félicité <deogracia@free.fr>

# PATH should only include /usr/* if it runs after the mountnfs.sh script
PATH=/sbin:/usr/sbin:/bin:/usr/bin
DESC=mailhog             # Introduce a short description here
NAME=mailhog             # Introduce the short server's name here
DAEMON=/usr/sbin/mailhog # Introduce the server's location here
DAEMON_ARGS=""             # Arguments to run the daemon with
PIDFILE=/var/run/$NAME.pid
SCRIPTNAME=/etc/init.d/$NAME

# Exit if the package is not installed
[ -x $DAEMON ] || exit 0

# Read configuration variable file if it is present
[ -r /etc/default/$NAME ] && . /etc/default/$NAME

# Load the VERBOSE setting and other rcS variables
. /lib/init/vars.sh

# Define LSB log_* functions.
# Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
. /lib/lsb/init-functions

#
# Function that starts the daemon/service
#
do_start()
{
	# Return
	#   0 if daemon has been started
	#   1 if daemon was already running
	#   2 if daemon could not be started
	start-stop-daemon --start --quiet --pidfile $PIDFILE --exec $DAEMON --test > /dev/null \
		|| return 1
	start-stop-daemon --start --quiet --pidfile $PIDFILE --exec $DAEMON -- \
		$DAEMON_ARGS $DAEMON_OPTS\
		|| return 2
	# Add code here, if necessary, that waits for the process to be ready
	# to handle requests from services started subsequently which depend
	# on this one.  As a last resort, sleep for some time.
}

#
# Function that stops the daemon/service
#
do_stop()
{
	# Return
	#   0 if daemon has been stopped
	#   1 if daemon was already stopped
	#   2 if daemon could not be stopped
	#   other if a failure occurred
	start-stop-daemon --stop --quiet --retry=TERM/30/KILL/5 --pidfile $PIDFILE --name $NAME
	RETVAL="$?"
	[ "$RETVAL" = 2 ] && return 2
	# Wait for children to finish too if this is a daemon that forks
	# and if the daemon is only ever run from this initscript.
	# If the above conditions are not satisfied then add some other code
	# that waits for the process to drop all resources that could be
	# needed by services started subsequently.  A last resort is to
	# sleep for some time.
	start-stop-daemon --stop --quiet --oknodo --retry=0/30/KILL/5 --exec $DAEMON
	[ "$?" = 2 ] && return 2
	# Many daemons don't delete their pidfiles when they exit.
	rm -f $PIDFILE
	return "$RETVAL"
}

case "$1" in
  start)
    [ "$VERBOSE" != no ] && log_daemon_msg "Starting $DESC " "$NAME"
    do_start
    case "$?" in
		0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
		2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
	esac
  ;;
  stop)
	[ "$VERBOSE" != no ] && log_daemon_msg "Stopping $DESC" "$NAME"
	do_stop
	case "$?" in
		0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
		2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
	esac
	;;
  status)
       status_of_proc "$DAEMON" "$NAME" && exit 0 || exit $?
       ;;
  restart|force-reload)
	#
	# If the "reload" option is implemented then remove the
	# 'force-reload' alias
	#
	log_daemon_msg "Restarting $DESC" "$NAME"
	do_stop
	case "$?" in
	  0|1)
		do_start
		case "$?" in
			0) log_end_msg 0 ;;
			1) log_end_msg 1 ;; # Old process is still running
			*) log_end_msg 1 ;; # Failed to start
		esac
		;;
	  *)
	  	# Failed to stop
		log_end_msg 1
		;;
	esac
	;;
  *)
	echo "Usage: $SCRIPTNAME {start|stop|status|restart|force-reload}" >&2
	exit 3
	;;
esac

:


debian/postrm.ex
#!/bin/sh
# postrm script for mailhog
#
# see: dh_installdeb(1)

set -e

# summary of how this script can be called:
#        * <postrm> `remove'
#        * <postrm> `purge'
#        * <old-postrm> `upgrade' <new-version>
#        * <new-postrm> `failed-upgrade' <old-version>
#        * <new-postrm> `abort-install'
#        * <new-postrm> `abort-install' <old-version>
#        * <new-postrm> `abort-upgrade' <old-version>
#        * <disappearer's-postrm> `disappear' <overwriter>
#          <overwriter-version>
# for details, see http://www.debian.org/doc/debian-policy/ or
# the debian-policy package


case "$1" in
    purge|remove|upgrade|failed-upgrade|abort-install|abort-upgrade|disappear)
    ;;

    *)
        echo "postrm called with unknown argument \`$1'" >&2
        exit 1
    ;;
esac

# dh_installdeb will replace this with shell code automatically
# generated by other debhelper scripts.

#DEBHELPER#

exit 0


debian/postinst.ex
#!/bin/sh
# postinst script for mailhog
#
# see: dh_installdeb(1)

set -e

# summary of how this script can be called:
#        * <postinst> `configure' <most-recently-configured-version>
#        * <old-postinst> `abort-upgrade' <new version>
#        * <conflictor's-postinst> `abort-remove' `in-favour' <package>
#          <new-version>
#        * <postinst> `abort-remove'
#        * <deconfigured's-postinst> `abort-deconfigure' `in-favour'
#          <failed-install-package> <version> `removing'
#          <conflicting-package> <version>
# for details, see http://www.debian.org/doc/debian-policy/ or
# the debian-policy package


case "$1" in
    configure)
    ;;

    abort-upgrade|abort-remove|abort-deconfigure)
    ;;

    *)
        echo "postinst called with unknown argument \`$1'" >&2
        exit 1
    ;;
esac

if [ $2 ]; then
  if [ -x /usr/sbin/invoke-rc.d ]; then
    /usr/sbin/invoke-rc.d mailhog start || true
  else
    /etc/init.d/mailhog start || true
  fi
fi

# dh_installdeb will replace this with shell code automatically
# generated by other debhelper scripts.

#DEBHELPER#

exit 0


restore-permission.bash
#!/usr/bin/env bash

# still a temporary solution till I use the upstream source
# to create my deb package

chmod 644 package/DEBIAN/conffiles
chmod 755 package/DEBIAN/postinst
chmod 755 package/DEBIAN/postrm
chmod 755 package/DEBIAN/preinst
chmod 755 package/DEBIAN/prerm
chmod 755 package/etc/
chmod 755 package/etc/default/
chmod 644 package/etc/default/mailhog
chmod 755 package/etc/init.d/
chmod 755 package/etc/init.d/mailhog
chmod 755 package/etc/init.d/mailhog
chmod 755 package/usr/
chmod 755 package/usr/share/
chmod 755 package/usr/share/doc/
chmod 755 package/usr/share/doc/mailhog/
chmod 644 package/usr/share/doc/mailhog/changelog.Debian.gz
chmod 644 package/usr/share/doc/mailhog/copyright
chmod 755 package/usr/share/lintian/
chmod 755 package/usr/share/lintian/overrides/
chmod 644 package/usr/share/lintian/overrides/mailhog
chmod 755 package/usr/share/man/
chmod 755 package/usr/share/man/man8/
chmod 644 package/usr/share/man/man8/mailhog.8.gz



debian/mailhog.default.ex
# Defaults for mailhog initscript
# sourced by /etc/init.d/mailhog
# installed at /etc/default/mailhog by the maintainer scripts

#
# This is a POSIX shell fragment
#

# Defaults values to MailHog
#  -api-bind-addr="0.0.0.0:8025": HTTP bind interface and port for API, e.g. 0.0.0.0:8025 or just :8025
MAILHOG_API_BIND_ADDR="0.0.0.0:8025"

#  -api-host="": API URL for MailHog UI to connect to, e.g. http://some.host:1234
MAILHOG_API_HOST=""

#  -cors-origin="": CORS Access-Control-Allow-Origin header for API endpoints
MAILHOG_CORS_ORIGIN=""

#  -hostname="mailhog.example": Hostname for EHLO/HELO response, e.g. mailhog.example
MAILHOG_HOSTNAME='""mailhog.example"

#  -invite-jim=false: Decide whether to invite Jim (beware, he causes trouble)
MAILHOG_INVITE_JIM="false"

#  -jim-accept=0.99: Chance of accept
MAILHOG_JIM_ACCEPT="0.99"

#  -jim-disconnect=0.005: Chance of disconnect
MAILHOG_JIM_DISCONNECT="0.005"

#  -jim-linkspeed-affect=0.1: Chance of affecting link speed
MAILHOG_JIM_LINKSPEED_AFFECT="0.01"

#  -jim-linkspeed-max=10240: Maximum link speed (in bytes per second)
MAILHOG_JIM_LINKSPEED_MAX="10240"

#  -jim-linkspeed-min=1024: Minimum link speed (in bytes per second)
MAILHOG_JIM_LINKSPEED_MIN="10240"

#  -jim-reject-auth=0.05: Chance of rejecting authentication (AUTH)
MAILHOG_JIM_REJECT_AUTH="0.05"

#  -jim-reject-recipient=0.05: Chance of rejecting a recipient (RCPT TO)
MAILHOG_JIM_REJECT_RECIPIENT="0.05"

#  -jim-reject-sender=0.05: Chance of rejecting a sender (MAIL FROM)
MAILHOG_JIM_REJECT_SENDER="0.05"

#  -mongo-coll="messages": MongoDB collection, e.g. messages
MAILHOG_MONGO_COLL="messages"

#  -mongo-db="mailhog": MongoDB database, e.g. mailhog
MAILHOG_DB="mailhog"

#  -mongo-uri="127.0.0.1:27017": MongoDB URI, e.g. 127.0.0.1:27017
MAILHOG_URI="127.0.0.1:27017"

#  -outgoing-smtp="": JSON file containing outgoing SMTP servers
MAILHOG_OUTGOING_SMTP=""

#  -smtp-bind-addr="0.0.0.0:1025": SMTP bind interface and port, e.g. 0.0.0.0:1025 or just :1025
MAILHOG_SMTP_BIND_ADDR="0.0.0.0:1025"

#  -storage="memory": Message storage: memory (default) or mongodb
MAILHOG_STORAGE="memory"

#  -ui-bind-addr="0.0.0.0:8025": HTTP bind interface and port for UI, e.g. 0.0.0.0:8025 or just :8025
MAILHOG_UI_BIND_ADDR="0.0.0.0:8025"

# Additional options that are passed to the Daemon.
DAEMON_OPTS=""


debian/mailhog.8.ex
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH MAILHOG 8 "March 31, 2015" "version v0.1.6"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
mailhog \- Web and API based SMTP testing 
.SH SYNOPSIS
.B mailhog
.RI [ options ]
.br
.SH DESCRIPTION
Inspired by MailCatcher, easier to install, built with Go, MailHog runs without installation on multiple platforms.
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics,
.\" respectively.
\fBmailhog\fP is an email testing tool for developers:

.SH OPTIONS
A summary of options and their default values is included below.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-api-bind-addr="0.0.0.0:8025"
HTTP bind interface and port for API, e.g. 0.0.0.0:8025 or just :8025
.TP
.B \-api-host=""
API URL for MailHog UI to connect to, e.g. http://some.host:1234
.TP
.B \-cors-origin=""
CORS Access-Control-Allow-Origin header for API endpoints
.TP
.B \-hostname="mailhog.example"
Hostname for EHLO/HELO response, e.g. mailhog.example
.TP
.B \-invite-jim=false
Decide whether to invite Jim (beware, he causes trouble)
.TP
.B \-jim-accept=0.99
Chance of accept
.TP
.B \-jim-disconnect=0.005
Chance of disconnect
.TP
.B \-jim-linkspeed-affect=0.1
Chance of affecting link speed
.TP
.B \-jim-linkspeed-max=10240
Maximum link speed (in bytes per second)
.TP
.B \-jim-linkspeed-min=1024
Minimum link speed (in bytes per second)
.TP
.B \-jim-reject-auth=0.05
Chance of rejecting authentication (AUTH)
.TP
.B \-jim-reject-recipient=0.05
Chance of rejecting a recipient (RCPT TO)
.TP
.B \-jim-reject-sender=0.05
Chance of rejecting a sender (MAIL FROM)
.TP
.B \-mongo-coll="messages"
MongoDB collection, e.g. messages
.TP
.B \-mongo-db="mailhog"
MongoDB database, e.g. mailhog
.TP
.B \-mongo-uri="127.0.0.1:27017"
MongoDB URI, e.g. 127.0.0.1:27017
.TP
.B \-outgoing-smtp=""
JSON file containing outgoing SMTP servers
.TP
.B \-smtp-bind-addr="0.0.0.0:1025"
SMTP bind interface and port, e.g. 0.0.0.0:1025 or just :1025
.TP
.B \-storage="memory"
Message storage: memory (default) or mongodb
.TP
.B \-ui-bind-addr="0.0.0.0:8025"
HTTP bind interface and port for UI, e.g. 0.0.0.0:8025 or just :8025
.SH SEE ALSO
https://github.com/mailhog/MailHog
.SH AUTHOR
mailhog was written by Ian Kent (http://iankent.uk).
.PP
This manual page was written by Lionel Félicité <deogracia@free.fr>,
for the Debian project (and may be used by others).
